File: sortPy.js


function findSmallest(arr) {

    let smallest = arr[0]
    let smallestIndex = 0;


    for (let i = 1; i < arr.length; i++) {

        if (arr[i] < smallest) {
            smallest = arr[i]
            smallestIndex = i;
        }
    }



    return smallestIndex

}





function selectionSort(arr) {
    let newArr = []


    while (arr.length > 0) {
        let smallest = findSmallest(arr);
        newArr.push(arr)
    }
}

File: sortPy.py

def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1,len(arr)):
      if(arr[i] < smallest):
         smallest = arr[i]
         smallest_index = i
    return smallest_index



def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
       smallest = findSmallest(arr)
       newArr.append(arr.pop(smallest))
    return newArr



print(selectionSort([22,1,32,4,1342,424]))
print(selectionSort([1,1,32,4,1342,424,4243433,4242,12,52]))